### ISSUE TITLE
Overhangs & perimeters in briged area

### ISSUE THREAD

**Issue by supermerill** [slic3r/Slic3r#4476]
Hi
I was creating some options to let me throw badly made stl for 3D printing and having it not fail that much.

1) I added to "Extra perimeters if needed" the reverse: it add perimeters if possible instead to drawing bridge when there are overhangs : it check if the next perimeter is over a not-supported area and it ask to add one more perimeter in this case (repeat).

2) I added a feature that remove the detected "bridge" area from the perimeter generator (only the area that can be bridged, not the overhangs). With this, no perimeters are draw in these area (see pics for a use-case)

### **Extra perimeters if needed :** 
![overhang_2](https://user-images.githubusercontent.com/6536403/42468066-09747ba8-83b4-11e8-9e97-a2353d607299.png)

### ** Avoid unsupported perimeters :**
![noperionbridgedarea](https://user-images.githubusercontent.com/6536403/42468103-2b7383d4-83b4-11e8-920e-2922b23aca0f.png)

### ** An other exemple of Extra perimeters & a combination of both :**
![overhang_3](https://user-images.githubusercontent.com/6536403/42468142-4ba311a6-83b4-11e8-894d-1582c54f828c.png)

![problem](https://user-images.githubusercontent.com/6536403/42468370-0c5f6da4-83b5-11e8-8431-4cc4a024aa90.png)

What do you think of it?
What do you think it should do? (with an image please)
Do you have some torture test?

I will do some print of these "torture objects" to compare my changes with the current version.

my torture test object source:
[test_overhang.zip](https://github.com/slic3r/Slic3r/files/2177492/test_overhang.zip)

**Comment by lordofhyphens** [comment]
@supermerill if you open up a PR, the appveyor build server will build your changes.

Here's the list of all of the appveyor builds:
https://ci.appveyor.com/project/lordofhyphens/slic3r/history

To get the artifact for just that build, you click into the individual build log and then click the "Artifacts" tab.

**Comment by VanessaE** [comment]
You may be interested in #3950 also, @supermerill

**Comment by supermerill** [comment]
@Tinchus2009 
You're right. My third test is just an impossible print without supports.
But for the first and the second, is it better?

**Comment by Tinchus2009** [comment]
Yes! Actually I think you are thinking smehow in the same thing @supermill than a coupe of us. You are thinking about bridges but somehow the idea you are thinking about is actually overhangs
Right now a missin parameter  to better deal with overhang would be the abilitu to control the overhang speed and the fan speed at overhangs. Others features requested in #3950 would be awesome but I know that would be a lot of work to :)

**Comment by supermerill** [comment]
Some tests:
wade with natural pla. Maybe a more opaque plastic should be better to see the differences.

![first object](https://user-images.githubusercontent.com/6536403/42537592-a0f81750-8495-11e8-9b3a-47fadd3e8b67.jpg)
![second object](https://user-images.githubusercontent.com/6536403/42537595-a2b365f4-8495-11e8-971b-f237dc138893.jpg)
![third object](https://user-images.githubusercontent.com/6536403/42537597-a3ee2314-8495-11e8-9d1e-3a114e50da20.jpg)

**Comment by Tinchus2009** [comment]
Please dont take my comment in a bad way, I really respect the effort you are doing. haveing said that: I dont really see any improvements in the print I see. From my point of view about what I see, is like you are trying to find a way to print areas that need supports. In your first pic you show an example of a perimeter belonginf to a hole. Well, that hole can be printed with no problem useing supports. I the pictures... I can not say much because they are not very clear to me, but what I can see is no really a big difference. I would also ask you like Vanessa ask you: to take a look at feature request #3950 because right now you are already in that issue. If you already made a code to separate bridge perimeter from perimeter geenrator, you could also make a code wich separates overhang perimeters, and then we could work on those "isolated overhang perimters" in order to control features like: their speed, the cooling on them, etc.
